Tests/ApolloTests/Cache/ReadWriteFromStoreTests.swift

```diff
@@ -609,4 +609,155 @@ class ReadWriteFromStoreTests: XCTestCase, CacheDependentTesting, StoreLoading {
+        "hero(episode:JEDI)": CacheReference(key: "1101"),
+        "hero(episode:EMPIRE)": CacheReference(key: "2001")
+      ],
+      "2001": [
+        "id": "2001",
+        "name": "R2-D2",
+        "__typename": "Droid",
+        "friends": [
+          CacheReference(key: "1101"),
+          CacheReference(key: "1003")
+        ]
+      ],
+      "1101": ["__typename": "Human", "name": "Luke Skywalker", "id": "1101", "friends": []],
+      "1002": ["__typename": "Human", "name": "Han Solo", "id": "1002", "friends": []],
+      "1003": ["__typename": "Human", "name": "Leia Organa", "id": "1003", "friends": []],
+    ])
+
+    //
+    // 2. Attempt to successfully read out the three queries
+    //
+
+    let readCompletedExpectation = expectation(description: "All hero objects read from cache")
+
+    store.withinReadTransaction({ transaction in
+      let query = HeroAndFriendsNamesWithIDsQuery(episode: .newhope)
+      let data = try transaction.read(query: query)
+
+      XCTAssertEqual(data.hero?.__typename, "Human")
+      XCTAssertEqual(data.hero?.name, "Han Solo")
+
+      let friendsNames = data.hero?.friends?.compactMap { $0?.name }
+      XCTAssertEqual(friendsNames, [])
+
+    }, completion: { newHopeResult in
+      XCTAssertSuccessResult(newHopeResult)
+
+      self.store.withinReadTransaction({ transaction in
+        let query = HeroAndFriendsNamesWithIDsQuery(episode: .jedi)
+        let data = try transaction.read(query: query)
+
+        XCTAssertEqual(data.hero?.__typename, "Human")
+        XCTAssertEqual(data.hero?.name, "Luke Skywalker")
+
+        let friendsNames = data.hero?.friends?.compactMap { $0?.name }
+        XCTAssertEqual(friendsNames, [])
+
+      }, completion: { jediResult in
+        XCTAssertSuccessResult(jediResult)
+
+        self.store.withinReadTransaction({ transaction in
+          let query = HeroAndFriendsNamesWithIDsQuery(episode: .empire)
+          let data = try transaction.read(query: query)
+
+          XCTAssertEqual(data.hero?.__typename, "Droid")
+          XCTAssertEqual(data.hero?.name, "R2-D2")
+
+          let friendsNames = data.hero?.friends?.compactMap { $0?.name }
+          XCTAssertEqual(friendsNames, ["Luke Skywalker", "Leia Organa"])
+
+        }, completion: { empireResult in
+          defer { readCompletedExpectation.fulfill() }
+
+          XCTAssertSuccessResult(empireResult)
+        })
+      })
+    })
+
+    self.waitForExpectations(timeout: Self.defaultWaitTimeout)
+
+    //
+    // 3. Remove all objects matching the pattern `100`
+    // - This will remove `1002` (Han Solo, hero for the .newhope episode)
+    // - This will remove `1003` (Leia Organa, friend of the hero in .empire episode)
+    //
+
+    let removeCompletedExpectation = expectation(description: "Hero objects removed from cache by pattern")
+
+    store.withinReadWriteTransaction({ transaction in
+      try transaction.removeObjects(matching: "100")
```

Comment:
I wonder if we should have another test to validate the case-insensitivity-by-default behaviour. 🤔

Reply:
I think that's probably a good idea if we're going to be saying it's case insensitive, especially to make sure the case insensitivity is preserved across both `InMemoryNormalizedCache` and `SQLiteNormalizedCache`

Reply:
See https://github.com/apollographql/apollo-ios/pull/1848/files#diff-15a6945fd0d80fdabb91e30e1b1ffb06c396e8a7466602afb574f13b2c5b650fR613